core/backend/src/IModelDb.ts

```diff
@@ -2167,9 +2173,12 @@ export namespace IModelDb {
         });
         return elProps.id = this._iModel[_nativeDb].insertElement(elProps, options);
       } catch (err: any) {
-        err.message = `Error inserting element [${err.message}]`;
-        err.metadata = { elProps };
-        throw err;
+        if (err.iTwinErrorId !== undefined) {
+          err.message = `Error inserting element [${err.message}]`;
+          err.metadata = { elProps };
+          throw err;
+        }
+        throw new IModelError(err.errorNumber, `Error inserting element [${err.message}]`, elProps);
```

Comment:
I don't think this is necessary or good. If there is any place in our native code where we don't include iTwinErrorId (why?), we should fix it. We should not add new uses of IModelError.